static/js/src/cve/cve.js

```diff
@@ -44,74 +36,22 @@ const ltsReleases = Object.values(ltsReleasesObj).map(
 const unmaintainedReleases = Object.values(unmaintainedReleasesObj).map(
   (release) => release.codename
 );
-let filterParams = [];
-
-function handleCveIdInput(value) {
-  const packageInput = document.querySelector("#package");
-  const priorityInput = document.querySelector("#priority");
-  const componentInput = document.querySelector("#component");
-  const statusInputs = document.querySelectorAll(".js-status-input");
-  const ubuntuVersionInputs = document.querySelectorAll(
-    ".js-ubuntu-version-input"
-  );
-  const addRowButtons = document.querySelectorAll(".js-add-row");
-  const removeRowButtons = document.querySelectorAll(".js-remove-row");
-  const searchButtonText = document.querySelector(".cve-search-text");
-  const searchButtonValidCveText = document.querySelector(
-    ".cve-search-valid-cve-text"
-  );
-
-  if (isValidCveId(value)) {
-    searchButtonValidCveText.classList.remove("u-hide");
-    searchButtonText.classList.add("u-hide");
-
-    disableField(packageInput);
-    disableField(priorityInput);
-    disableField(componentInput);
-
-    statusInputs.forEach((statusInput) => disableField(statusInput));
-    ubuntuVersionInputs.forEach((ubuntuVersionInput) =>
-      disableField(ubuntuVersionInput)
-    );
-    addRowButtons.forEach((addRowButton) => disableField(addRowButton));
-    removeRowButtons.forEach((removeRowButton) =>
-      disableField(removeRowButton)
-    );
-  } else {
-    enableField(packageInput);
-    enableField(priorityInput);
-    enableField(componentInput);
-
-    statusInputs.forEach((statusInput) => enableField(statusInput));
-    ubuntuVersionInputs.forEach((ubuntuVersionInput) =>
-      enableField(ubuntuVersionInput)
-    );
-    removeRowButtons.forEach((removeRowButton, index) => {
-      if (index > 0) {
-        enableField(removeRowButton);
-      }
-    });
-
-    searchButtonText.classList.remove("u-hide");
-    searchButtonValidCveText.classList.add("u-hide");
-  }
-}
-
-handleCveIdInput(searchInput.value);
 
 function handleSearchInput(event) {
   if (event.key === "Enter") {
     searchForm.submit();
-  } else {
-    handleCveIdInput(event.target.value);
   }
 }
-
 searchInput.addEventListener("keyup", handleSearchInput);
 
-attachEvents();
-handleButtons();
-disableSelectedVersions();
+// Adds listener to package filter text field and adds value
+// to URLSearchParams object
+function handlePackageInput() {
+  packageInput.addEventListener("input", function (event) {
+    urlParams.set("package", packageInput.value);
+  });
+}
+handlePackageInput();
 
 // Adds event listeners to all filter checkboxes
 function handleFilters() {
```

Comment:
Could it be helpful to add this check before line 60 to tick all the maintained releases' checkboxes when **All maintained releases** is selected? This way we can reduce the need to loop through the array of maintained releases as we add/remove params.

```
if (checkbox.value === "maintained") {
  maintainedReleases.forEach(function (release) {
    const checkbox = getCheckboxFromRelease(release);
    checkbox?.checked = event.target.checked;
  });
}
```

```
function getCheckboxFromRelease(release) {
  const releaseCheckboxesArray = Array.from(releaseCheckboxes);
  const checkbox = releaseCheckboxesArray.find(checkbox => checkbox.value === release);

  return checkbox;
}
```

Reply:
Great catch, thanks!